cognite/extractorutils/configtools.py

```diff
@@ -503,9 +504,14 @@ def load_yaml(
         InvalidConfigError: If any config field is given as an invalid type, is missing or is unknown
     """
 
-    def env_constructor(_, node):
-        # Expnadvars uses same syntax as our env var substitution
-        return os.path.expandvars(node.value)
+    def env_constructor(l: yaml.SafeLoader, node):
+        expanded_value = os.path.expandvars(node.value)
+        new_node = copy.deepcopy(node)
+        new_node.value = expanded_value
+        return l.construct_yaml_bool(new_node)
```

Comment:
It coult be just fine to do this maybe 🤔 Wasn't sure if we want to modify the original node but deepcopy is also not really nice. I don't like this solution as well because it will only work for string and bool environment variable conversion but couldn't find a way to enforce our constructor to run first.  @mathialo 
```suggestion
    def env_constructor(l: yaml.SafeLoader, node):
        expanded_value = os.path.expandvars(node.value)
        node.value = expanded_value
        return l.construct_yaml_bool(new_node)
```